activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -112,20 +112,27 @@ def table(table, stream)
           tbl = StringIO.new
 
           # first dump primary key column
-          pk = @connection.primary_key(table)
+          pk = @connection.primary_keys(table) rescue nil
+          if pk.nil?
+            pk = @connection.primary_key(table)
+          else
+            pk = pk.first unless pk.size > 1
+          end
 
           tbl.print "  create_table #{remove_prefix_and_suffix(table).inspect}"
-          pkcol = columns.detect { |c| c.name == pk }
-          if pkcol
-            if pk != 'id'
-              tbl.print %Q(, primary_key: "#{pk}")
-            end
+
+          case pk
+          when String
+            tbl.print ", primary_key: #{pk.inspect}" unless pk == 'id'
+            pkcol = columns.detect { |c| c.name == pk }
             pkcolspec = @connection.column_spec_for_primary_key(pkcol)
             if pkcolspec
               pkcolspec.each do |key, value|
                 tbl.print ", #{key}: #{value}"
               end
             end
+          when Array
+            tbl.print ", primary_key: #{pk.inspect}, id: false"
```

Comment:
`column_spec_for_primary_key` was added by #18228.
This is for single column primary key of custom type and options (especially for unsigned support! #17696).
This is no need for composite primary key.